@teharris1 The test failure looks it might be an actual problem here:

```
FAILED tests/components/insteon/test_init.py::test_setup_entry[pyloop] - AttributeError: 'Hub' object has no attribute 'read_write_mode'
FAILED tests/components/insteon/test_init.py::test_import_plm[pyloop] - AttributeError: 'Hub' object has no attribute 'read_write_mode'
FAILED tests/components/insteon/test_init.py::test_import_hub1[pyloop] - AttributeError: 'Hub' object has no attribute 'read_write_mode'
FAILED tests/components/insteon/test_init.py::test_import_hub2[pyloop] - AttributeError: 'Hub' object has no attribute 'read_write_mode'
FAILED tests/components/insteon/test_init.py::test_import_options[pyloop] - AttributeError: 'Hub' object has no attribute 'read_write_mode'
```

--- a/homeassistant/components/insteon/__init__.py
+++ b/homeassistant/components/insteon/__init__.py
@@ -49,7 +49,7 @@ async def async_get_device_config(hass, config_entry):
         with suppress(AttributeError):
             await devices[address].async_status()
 
-    load_aldb = devices.modem.read_write_mode == ReadWriteMode.UNKNOWN
+    load_aldb = devices.modem.aldb.read_write_mode == ReadWriteMode.UNKNOWN
     await devices.async_load(id_devices=1, load_modem_aldb=load_aldb)
     for addr in devices:
         device = devices[addr]

